src/consensus/pbft/pbft.h

```diff
@@ -372,11 +478,110 @@ namespace pbft
       switch (serialized::peek<PbftMsgType>(data, size))
       {
         case pbft_message:
+        {
           serialized::skip(data, size, sizeof(PbftHeader));
           message_receiver_base->receive_message(data, size);
           break;
-        default:
-        {}
+        }
+        case pbft_append_entries:
+        {
+          AppendEntries r;
+
+          auto append_entries_index = store->current_version();
+
+          try
+          {
+            r =
+              channels->template recv_authenticated<AppendEntries>(data, size);
+          }
+          catch (const std::logic_error& err)
+          {
+            LOG_FAIL_FMT(err.what());
+            return;
+          }
+
+          LOG_TRACE_FMT(
+            "Append entries message from {}, my ae index is {}",
+            r.from_node,
+            append_entries_index);
+
+          auto node = nodes.find(r.from_node);
+          if (node != nodes.end())
+          {
+            node->second = r.idx;
+          }
+          else
+          {
+            nodes[r.from_node] = r.idx;
+          }
+
+          if (r.idx <= append_entries_index)
+          {
+            LOG_TRACE_FMT(
+              "Skipping append entries msg for index {} as we are at index {}",
+              r.idx,
+              append_entries_index);
+            break;
+          }
+
+          for (Index i = r.prev_idx + 1; i <= r.idx; i++)
+          {
+            append_entries_index = store->current_version();
+            LOG_TRACE_FMT("Recording entry for index {}", i);
+
+            if (i <= append_entries_index)
+            {
+              // If the current entry has already been deserialised, skip the
+              // payload for that entry
+              LOG_INFO_FMT(
+                "Skipping index {} as we are at index {}",
+                i,
+                append_entries_index);
+              ledger->skip_entry(data, size);
+              continue;
+            }
+
+            auto ret = ledger->get_entry(data, size);
+
+            if (!ret.second)
+            {
+              // NB: This will currently never be triggered.
+              // This should only fail if there is malformed data. Truncate
+              // the log and reply false.
+              LOG_FAIL_FMT(
+                "Recv append entries to {} from {} but the data is malformed",
+                local_id,
+                r.from_node);
+              ledger->truncate(r.prev_idx);
+              return;
+            }
+
+            ccf::Store::Tx tx;
+            auto deserialise_success = store->deserialise_views(
```

Comment:
Minor: no need for `deserialise_success` intermediate variable.

Reply:
I think I like it better with the intermediate variable, it makes it clearer that we are checking for the success or failure of the deserialization